### PR TITLE
Add data science notebook image

### PR DESCRIPTION
**Pull request by atheo89**
Added data science notebook image

## Description
Files that have been added:
- Dockerfile
- Kustomize objects
- requirment.in 

Files that have been modified:
- Makefile

## How Has This Been Tested?
`podman run -it --rm -p 8888:8888 quay.io/{USER_REGISTRY}:jupyter-data-science-ubi8-python-3.8`

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: jupyter/datascience/ubi8-python-3.8/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    authoritative-source-url="https://github.com/opendatahub-io/notebooks" \
+    io.openshift.build.commit.ref="main" \
+    io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/jupyter/datascience/ubi8-python-3.8" \
+    io.openshift.build.image="quay.io/opendatahub/notebooks:jupyter-datascience-ubi8-python-3.8"
+
+COPY requirements.txt ./
+
+# Fix permissions to support pip in Openshift environments
+RUN chmod -R g+w /opt/app-root/lib/python3.8/site-packages && \
+      fix-permissions /opt/app-root -P
```
Comment (samuelvl) [inline jupyter/datascience/ubi8-python-3.8/Dockerfile:18]: This should happen after `pip install`

File: jupyter/datascience/ubi8-python-3.8/requirements.in [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# JupyterLab extensions
+jupyter-resource-usage==0.6.1
+jupyterlab-s3-browser==0.10.1
+jupyterlab-widgets==3.0.3
+jupyterlab-lsp==3.10.0
+jupyterlab-git==0.30.0
+jupyter-bokeh==3.0.4
+plotly==5.5.0
+nbdime==3.0.0
```
Comment (samuelvl) [inline jupyter/datascience/ubi8-python-3.8/requirements.in:18]: `jupyterlab-git` and `nbdime` are already included in the minimal notebook image, no need to include them twice

File: jupyter/datascience/ubi8-python-3.8/requirements.in [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# JupyterLab data science requirements
+scikit-learn==0.24.1
+kafka-python==2.0.2
+matplotlib==3.4.1
+pandas==1.2.4
+numpy==1.19.2
+boto3==1.17.11
+scipy==1.6.2
+
+# JupyterLab extensions
```
Comment (samuelvl) [inline jupyter/datascience/ubi8-python-3.8/requirements.in:10]: Some extensions from the ODS CI framework are missing. See https://github.com/red-hat-data-services/ods-ci/blob/master/tests/Tests/500__jupyterhub/plugin-verification.robot#L17

File: Makefile [hunk 4]
```diff
@@ -67,6 +67,14 @@ jupyter-minimal-ubi8-python-3.8: base-ubi8-python-3.8
 		jupyter/minimal/ubi8-python-3.8/requirements.txt)
 	$(call image,jupyter/minimal/ubi8-python-3.8,base/ubi8-python-3.8)
 
+# Build and push jupyter-datascience-ubi8-python-3.8 image to the registry
+.PHONY: jupyter-datascience-ubi8-python-3.8
+jupyter-datascience-ubi8-python-3.8: jupyter-minimal-ubi8-python-3.8
```
Comment (samuelvl) [inline Makefile:72]: Documentation in README.md is missing for this new notebook

**Review by samuelvl — CHANGES_REQUESTED** [review]

**Comment by atheo89** [comment]
Thanks for the review! I rebased my local repo to apply your comments. About the extensions from the ODS CI framework, the `jupyterlab_requirments` is missing on the old minimal image. I have added it now.

File: Makefile [hunk 5]
```diff
@@ -67,6 +67,14 @@ jupyter-minimal-ubi8-python-3.8: base-ubi8-python-3.8
...
+	$(call pip_compile,bootstrap/ubi8-python-3.8,\
+		base/ubi8-python-3.8/requirements.in jupyter/datascience/ubi8-python-3.8/requirements.in,\
```
Comment (samuelvl) [inline Makefile:74]: We have to include `jupyter/minimal/ubi8-python-3.8/requirements.in` too.

**Review by samuelvl — CHANGES_REQUESTED** [review]

**Comment by samuelvl** [comment]
@atheo89 

> About the extensions from the ODS CI framework, the jupyterlab_requirments is missing on the old minimal image. I have added it now.

What do you mean? I can't find the `jupyterlab_requirements` package in the old minimal notebook:

```shell
$ podman run --rm quay.io/modh/odh-minimal-notebook-container:v1.17.0-9 pip show jupyterlab-requirements
WARNING: Package(s) not found: jupyterlab-requirements

$ podman run --rm quay.io/modh/odh-minimal-notebook-container:v1.17.0-9 pip show jupyterlab_requirements
WARNING: Package(s) not found: jupyterlab_requirements
```

**Comment by atheo89** [comment]
Sorry, I wasn't clear before, I added the `jupyterlab_requirements` into the [requirements.in ](https://github.com/atheo89/notebooks/blob/02586a49f3ca3217754db22638b7359d6c23257e/jupyter/datascience/ubi8-python-3.8/requirements.in#L11) in the new data science notebook image because it exists into [ods-ci](https://github.com/red-hat-data-services/ods-ci/blob/master/tests/Tests/500__jupyterhub/plugin-verification.robot#L17) Should I leave it or exclude it?

File: README.md [hunk 6]
```diff
@@ -51,6 +52,24 @@ You can use a different registry by overwriting the `IMAGE_REGISTRY` variable:
...
 ```
 
+### Building jupyter-datascience-ubi8-python-3.8 image
+
+The jupyter-datascience-ubi8-python-3.8 notebook image includes a list of generic datascience packages.
+
+Build the `jupyter-datascience-ubi8-python-3.8` by running the following command:
+
+```shell
+make jjupyter-datascience-ubi8-python-3.8
```
Comment (samuelvl) [inline README.md:62]: Small typo `s/jjupyter/jupyter`

**Pull request merged**